include/execution.hpp

```diff
@@ -2461,7 +2461,110 @@ namespace _P2300::execution {
   /////////////////////////////////////////////////////////////////////////////
   // [execution.senders.adaptors.bulk]
   namespace __bulk {
+    template <class _ReceiverId, integral _Shape, class _FunId>
+      class __receiver
+        : receiver_adaptor<__receiver<_ReceiverId, _Shape, _FunId>, __t<_ReceiverId>> {
+        using _Receiver = __t<_ReceiverId>;
+        using _Fun = __t<_FunId>;
+        friend receiver_adaptor<__receiver, _Receiver>;
+
+        [[no_unique_address]] _Shape __shape_;
+        [[no_unique_address]] _Fun __f_;
+
+        template <class... _As>
+        void set_value(_As&&... __as) && noexcept 
+          requires __nothrow_invocable<_Fun, _Shape, _As...> {
+          for (_Shape __i{}; __i != __shape_; ++__i) {
+            std::invoke(__f_, __i, std::forward<_As>(__as)...);
```

Comment:
Since `_Shape` is currently required to model `integral`, `__f_` cannot be a member pointer, and `std::invoke` can be replaced with an ordinary function call. I avoid `std::invoke` when I know it's not needed for improved compile times and diagnostics. (Can also replace the constraint with `__nothrow_callable`.)

Also, it looks suspicious to me that we're `std::forward`-ing the values into the callable. We don't want to let the callable consume the value. That would be bugged.